apps/api-extractor/src/collector/Collector.ts

```diff
@@ -306,6 +307,43 @@ export class Collector {
     return this._entitiesByAstEntity.get(astEntity);
   }
 
+  /**
+   * Returns the name to emit for the given `symbol`.
+   * This is usually the `name` of that `symbol`.
+   * Only if this `symbol` has a `CollectorEntity`, the `nameForEmit` of that entity is used.
+   * @param symbol the symbol to compute the name to emit for
+   * @returns the name to emit for the given `symbol`
+   */
+  public getEmitName(symbol: ts.Symbol): string {
+    const collectorEntity: CollectorEntity | undefined = this._entitiesBySymbol.get(symbol);
+    return collectorEntity?.nameForEmit ?? symbol.name;
+  }
+
+  public getParentSymbols(symbol: ts.Symbol): ts.Symbol[] {
+    const result: ts.Symbol[] = [];
+    let currentSymbol: ts.Symbol | undefined = symbol;
+    while (currentSymbol) {
+      currentSymbol = this._getParentSymbol(currentSymbol);
+      if (currentSymbol) {
+        result.unshift(currentSymbol);
+      }
+    }
+    return result;
+  }
+
+  private _getParentSymbol(symbol: ts.Symbol): ts.Symbol | undefined {
+    const collectorEntity: CollectorEntity | undefined = this._entitiesBySymbol.get(symbol);
+    if (collectorEntity) {
+      const firstNamespaceImport: AstNamespaceImport | undefined = [
+        ...collectorEntity.astNamespaceImports
+      ][0];
+      if (firstNamespaceImport) {
+        return (firstNamespaceImport.declaration as unknown as ts.Type).symbol;
```

Comment:
Yes, this is the code I was referring to somewhere in my length commit message (should probably have been in the code):

> Namespace imports are now also registered in the
_entitiesBySymbol Map, but I did not find the API to
access the symbol of the AstNamespaceImport astEntity
(ugly type assertion 'as unknown as ts.Type', which is
probably not even the right type to access 'symbol').
I found the reference while debugging, but could not
find it in any meaningful subtype.

I also remember seeing a `moduleSymbol` during debugging, but iirc, it did not point to the desired symbol.
Did you find out where `symbol` comes from? Is it maybe meant to be "private"? Or is there maybe a gap in the type definition of the AST declaration data structure?

Reply:
I think this is the only remaining open issue which is still present in my rebased branch. So it would be great if you helped me find a better way to retrieve this namespace declaration symbol.

Reply:
Resolved by my last commit https://github.com/microsoft/rushstack/pull/3602/commits/032e4014add5939675e5a1be1b96286c2d08099c.